web-registry/src/components/templates/ProjectDetails/hooks/useMoreProjects.ts

```diff
@@ -0,0 +1,56 @@
+import { useMemo } from 'react';
+import shuffle from 'lodash/shuffle';
+
+import { ProjectWithOrderData } from 'pages/Projects/Projects.types';
+import { useProjectsWithOrders } from 'hooks/projects/useProjectsWithOrders';
+
+const PROJECTS_LIMIT = 3;
+
+export default function useMoreProjects(
+  projectId: string,
+): ProjectWithOrderData[] {
+  // get normalized projects with sell order data
+  const { projectsWithOrderData } = useProjectsWithOrders({});
```

Comment:
`useProjectsWithOrders` is used without a limit here, which means that it will make an individual query to get the metadata for each project while we only need 3 projects.
The logic that picks the 3 random projects should be placed before we do the metadata fetch request to avoid doing unnecessary requests. By doing so we would drastically reduce the number of queries done by the hook.

Reply:
This could be a new param to give for the hook:
`useProjectsWithOrders({limit: 3, shuffle: true})`
or it could something more generic to fit more use cases:
`useProjectsWithOrders({limit: 3, sortFunction: () => ...})`
The idea would be to pass a sort function that puts you 3 random projects on top of the array, and then the limit would slice only those 3 projects and it would make only 3 metadata requests.

Reply:
I have tried to improve a bit this point, although not so generic. I hope it works for now. Now we make the fair and necessary requests of the projects that we want to show. On the other hand it seems that all these requests were being done twice due to some rerender or something..